### PR TITLE
Davide/tutorial

### PR DESCRIPTION
**Pull request by aittalam**
Fixed missing lib and updated docs + examples for a prometheus tutorial.

### CODE REVIEW

File: docs/source/evaluation_concepts.md [hunk 1]
```diff
@@ -1,6 +1,54 @@
...
+    wandb.log_artifact(artifact)
+```
+
+In the `evaluation` directory, you will find a sample `prometheus_config.yaml` file for running Prometheus evaluation. Before using it, you will need to specify the `path` of the input dataset, the `base_url` where the Prometheus model is served, and
+the `tracking` options to save the evaluation output on wandb.
+
+You can then run the evaluation as:
+
+```
+python -m lm_buddy evaluate prometheus --config /path/to/prometheus_config.yaml
```
Comment (sfriedowitz) [inline docs/source/evaluation_concepts.md:53]:
```suggestion
lm_buddy evaluate prometheus --config /path/to/prometheus_config.yaml
```

assuming that its installed in your python environment

File: docs/source/evaluation_concepts.md [hunk 2]
```diff
@@ -1,6 +1,54 @@
 Evaluation
 ====================================
 
-All evaluation is currently done via [EleutherAI's lm-evaluation-harness package](https://github.com/EleutherAI/lm-evaluation-harness) run as a process. Evaluation can either happen on HuggingFace models hosted on the Hub, or on local models in shared storage on a Linux filesystem that resolve to [Weights and Biases Artifacts](https://docs.wandb.ai/ref/python/artifact) objects.
+## lm-evaluation-harness
+
+Evaluation is currently done via [EleutherAI's lm-evaluation-harness package](https://github.com/EleutherAI/lm-evaluation-harness) run as a process. Evaluation can either happen on HuggingFace models hosted on the Hub, or on local models in shared storage on a Linux filesystem that resolve to [Weights and Biases Artifacts](https://docs.wandb.ai/ref/python/artifact) objects
```
Comment (sfriedowitz) [inline docs/source/evaluation_concepts.md:6]:
```suggestion
[EleutherAI's lm-evaluation-harness package](https://github.com/EleutherAI/lm-evaluation-harness) is used internally to access a variety of benchmark datasets. The model to evaluate can be loaded directly from the HuggingFace Hub, from a local model checkpoint saved on the filesystem, or from a [Weights and Biases artifact](https://docs.wandb.ai/ref/python/artifact) object based on the `path` parameter specified in the evaluation config. 
```

This previous statement didn't quite make sense, so editing it for accuracy.

**Review by sfriedowitz — COMMENTED** [review]
couple quick comments but the new stuff looks fine

File: docs/source/evaluation_concepts.md [hunk 3]
```diff
@@ -1,6 +1,54 @@
...
 
 In the `evaluation` directory, there are sample files for running evaluation on a model in HuggingFace (`lm_harness_hf_config.yaml`), or using a local inference server hosted on vLLM, (`lm_harness_inference_server_config.yaml`).
+
+## Prometheus
+
+Evaluation relies on [Prometheus](https://github.com/kaistAI/Prometheus) as LLM judge. We internally serve it via [vLLM](https://github.com/vllm-project/vllm) but any other OpenAI API compatible service should work (e.g. llamafile via their `api_like_OAI.py` script).
+
+Input datasets _must_ be in HuggingFace format. The code below shows how to convert Prometheus benchmark datasets and optionally save them as wandb artifacts:
```
Comment (veekaybee) [inline docs/source/evaluation_concepts.md:14]: suggestion: might want to clarify what huggingface format is, in this case - loading via the `dataset` library call? `dataset = load_dataset("PolyAI/minds14", "en-US", split="train")`

**Review by veekaybee — APPROVED** [review]

File: docs/source/evaluation_concepts.md (hunk 1 above)
Comment (aittalam) [reply docs/source/evaluation_concepts.md]: Good catch, thanks!

**Comment by aittalam** [comment]
Just a heads up that I got a `argument of type 'PosixPath' is not iterable` error from a test in `tests/unit/test_preprocessing.py`. Fixed it by casting on str but was surprised to see that happening.

**Pull request merged**